ldm/simplet2i.py

```diff
@@ -522,17 +529,12 @@ def _new_seed(self):
         self.seed = random.randrange(0, np.iinfo(np.uint32).max)
         return self.seed
 
-    def _get_device(self):
-        device_type = choose_torch_device()
-        return torch.device(device_type)
-
     def load_model(self):
         """Load and initialize the model from configuration variables passed at object creation time"""
         if self.model is None:
             seed_everything(self.seed)
             try:
                 config      = OmegaConf.load(self.config)
-                self.device = self._get_device()
```

Comment:
moved this logic to `__init__`

Reply:
Ok, I finally understand this pull request. [Line 374](https://github.com/lstein/stable-diffusion/blob/4b560b50c21dcf67872753d11410671f5b48531a/ldm/simplet2i.py#L374
) has a cuda call that needs `if torch.cuda.is_available():` also. Do you want to add that commit to this also? I'm new to using GitHub for collaboration so I'm not sure if I can do it.

[Line 547](https://github.com/lstein/stable-diffusion/blob/4b560b50c21dcf67872753d11410671f5b48531a/ldm/simplet2i.py#L547) mentions cuda, and I believe that is incorrect information now.

Lastly, are they getting the device *after* they load the model for a reason? Otherwise, I don't see any reason to not move it to __init__.

Reply:
I'm new to PyTorch, but to my eyes `_get_device` doesn't seem to take the model into consideration at all. I'll take a look at the other place that calls cuda and add a commit.

Reply:
> '%4.2fG' % (torch.cuda.max_memory_allocated() / 1e9),

OK, I didn't wrap this in a `if torch.cuda.is_available()` because it didn't cause a missing attribute exception or anything that would impact the functionality, it just ends up printing 0. I'll change it if we can figure out a way to report the actual memory allocated for MPS.

Reply:
I'm not sure what to do about https://github.com/lstein/stable-diffusion/blob/4b560b50c21dcf67872753d11410671f5b48531a/ldm/simplet2i.py#L547 even when that was correct, it seems overly broad to print that error message for any AttributeError. I'd opt for just removing that message all together and just print the exception message instead.

Reply:
I added a commit for the error message.